src/ForeignController.sol

```diff
@@ -137,7 +155,7 @@ contract ForeignController is AccessControl {
     /**********************************************************************************************/
 
     function depositPSM(address asset, uint256 amount)
-        external onlyRole(RELAYER) isActive returns (uint256 shares)
+        external onlyRole(RELAYER) isActive rateLimitedAsset(LIMIT_PSM_DEPOSIT, asset, amount) returns (uint256 shares)
```

Comment:
Shouldn't depositPSM/withdrawPSM be opposite actions that can add to the limit of the other one? Manipulating PSM proportions is going to be relatively easy and the DOSing threat is very much a thing here.

Reply:
In this case I think we want them separate to protect against different issues which I mentioned above. This is not the same as mainnet PSM where there is really no limit to the amount of USDC that can be swapped to USDS because we consider USDC to be $1 as a core assumption of the protocol.